File: Practice/Tree/Tree.py
class Node:
    def __init__(self, data):
        self.left = None
        self.right = None
        self.data = data


"""
           7
          /   \
        5       8
      /   \    /  \
     4     3  9    10
"""


def setPerfectTree():
    root = Node(7)
    root.left = Node(5)
    root.right = Node(8)
    root.left.left = Node(4)
    root.left.righ = Node(3)
    root.right.left = Node(9)
    root.right.right = Node(10)

    return root


"""
            7
          /   \
        5       8
      /   \    /  \
     4     3  9    10
    /  \
    2   1
"""


def setCompleteTree():
    root = Node(7)
    root.left = Node(5)
    root.right = Node(8)
    root.left.left = Node(4)
    root.left.right = Node(3)
    root.left.left.left = Node(2)
    root.left.left.right = Node(1)
    root.right.left = Node(9)
    root.right.right = Node(10)

    return root


"""
            5
          /   \
        1       3
      /   \    /  \
     2     4  7     10
             /  \
            8    9
"""


def setPerfectTree():
    root = Node(5)
    root.left = Node(1)

    root.left.left = Node(2)
    root.left.right = Node(4)

    root.right = Node(3)
    root.right.right = Node(10)
    root.right.left = Node(7)
    root.right.left.left = Node(8)
    root.right.left.right = Node(9)

    return root

File: Practice/Tree/Tree.java
package Practice.Tree;

class Node {
    int key;
    Node left, right;

    public Node(int data) {
        key = data;
        left = right = null;
    }

    public static void main(String[] args) {
        System.out.println();
    }
}

class Tree {
    Node root;

    Tree(int key) {
        root = new Node(key);
    }

    Tree() {
        root = null;
    }

    Tree setPerfectTree() {
        Tree tree = new Tree();
        tree.root = new Node(7);
        tree.root.left = new Node(5);
        tree.root.left.left = new Node(4);
        tree.root.left.right = new Node(3);

        tree.root.right = new Node(8);
        tree.root.right.left = new Node(9);
        tree.root.right.right = new Node(20);

        return tree;
    }

    Tree setCompleteTree() {
        Tree tree = new Tree();
        tree.root = new Node(7);
        tree.root.left = new Node(5);
        tree.root.left.left = new Node(4);
        tree.root.left.left.left = new Node(2);
        tree.root.left.left.right = new Node(1);

        tree.root.left.right = new Node(3);
        tree.root.left.right.left = new Node(7);
        tree.root.left.right.right = new Node(10);
        tree.root.left.right.left.left = new Node(8);
        tree.root.left.right.left.right = new Node(9);

        return tree;

    }

    Tree setFullTree() {
        Tree tree = new Tree();
        tree.root = new Node(5);
        tree.root.left = new Node(1);
        tree.root.left.left = new Node(2);
        tree.root.left.right = new Node(4);

        tree.root.right = new Node(3);
        tree.root.right.left = new Node(7);
        tree.root.right.left.left = new Node(8);
        tree.root.right.left.right = new Node(9);

        tree.root.right.right = new Node(10);

        return tree;
    }

    public static void main(String[] args) {
        System.out.println();
    }
}